compose-ui/src/androidMain/kotlin/com/example/myapplication/root/RootPreview.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.myapplication.root
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import com.arkivanov.decompose.DefaultComponentContext
+import com.arkivanov.essenty.lifecycle.LifecycleRegistry
+import com.example.myapplication.shared.root.DefaultRootComponent
+
+/**
+ * This preview can support navigation.
+ * (Use the default Android Studio @Preview function).
+ */
+@Preview
+@Composable
+fun RootPreview() {
+  RootContent(
+    DefaultRootComponent(
```

Comment:
Using real implementations in previews may cause problems, and it's also not possible to fake data for previews. It would be nice to create `PreviewRootComponent`, see an example [here](https://github.com/arkivanov/Decompose/blob/master/sample/shared/shared/src/commonMain/kotlin/com/arkivanov/sample/shared/tabs/PreviewTabsComponent.kt) of how we can fake the child stack.

Reply:
I've added the fake `PreviewRootComponent` as you said, but that'll make RootPreview same as MainPreview, with no navigation.
However I'm leaving this RootPreview in this androidMain source set for demonstration purpose. (e.g. testing out interactive widgets)

Reply:
> However I'm leaving this RootPreview in this androidMain source set for demonstration purpose. (e.g. testing out interactive widgets)

Since all the previews in the template are not interactive, I would rather prefer put all previews in `commonMain` and don't use Jetpack Compose previews in the template (i.e. don't use `buildFeatures {}` configuration and `uiTooling` dependency). One can easily setup this for their own needs.

Reply:
I got it.
But since `RootContent` doesn't have essentially any UI, could we just drop its preview entirely?